<a href="https://colab.research.google.com/github/Sandwhaletree/Mai_Finlab/blob/main/A_%E7%94%9F%E6%B4%BB%E6%87%89%E7%94%A8/C4_104%261111_%E8%81%B7%E7%BC%BA%E7%88%AC%E8%9F%B2_230919.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **🔥🔥🔥 104&1111求職爬蟲**

In [ ]:
#安裝selenium、chromium-chromedriver驅動及更新
!pip install selenium
!apt-get update
!apt install chromium-chromedriver

import requests
from bs4 import BeautifulSoup
import re
import random
import time
from urllib.parse import quote
from tqdm import tqdm, trange
import csv
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
import pandas as pd
from datetime import date
from IPython.display import clear_output
import os,os.path

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 6.0 MB 9.0 MB/s 
     |████████████████████████████████| 140 kB 49.5 MB/s 
     |████████████████████████████████| 384 kB 50.5 MB/s 
     |████████████████████████████████| 58 kB 6.6 MB/s 
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.24.3
    Uninstalling urllib3-1.24.3:
      Successfully uninstalled urllib3-1.24.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
requests 2.23.0 requires urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1, but you have urllib3 1.26.13 which is incompatible.
Get:1 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]
Ign:2 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Get:3 https://developer.downlo

/usr/local/lib/python3.7/dist-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)


In [ ]:
!ls

drive  sample_data


In [ ]:
def get_todate(): # 取檔名用的
    return date.today()

def selenium_get_Code_104(url):
    #chrome_options = Options() # 啟動無頭模式
    chrome_options = webdriver.ChromeOptions()
    chrome_options.add_argument('--headless')
    chrome_options.add_argument('--no-sandbox')
    chrome_options.add_argument('--disable-dev-shm-usage')
    driver = webdriver.Chrome('chromedriver',options=chrome_options)
    driver.get(url)
    save = driver.page_source
    driver.quit()#關閉瀏覽器
    soup = BeautifulSoup(save, "html.parser")
    page = soup.select('.page-select.js-paging-select.gtm-paging-top')[0].find_all('option')[-1].get('value')
    return page

def read_url(url):
    USER_AGENT_LIST = [
        "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; AcooBrowser; .NET CLR 1.1.4322; .NET CLR 2.0.50727)",
        "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0; Acoo Browser; SLCC1; .NET CLR 2.0.50727; Media Center PC 5.0; .NET CLR 3.0.04506)",
        "Mozilla/4.0 (compatible; MSIE 7.0; AOL 9.5; AOLBuild 4337.35; Windows NT 5.1; .NET CLR 1.1.4322; .NET CLR 2.0.50727)",
        "Mozilla/5.0 (Windows; U; MSIE 9.0; Windows NT 9.0; en-US)",
        "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Win64; x64; Trident/5.0; .NET CLR 3.5.30729; .NET CLR 3.0.30729; .NET CLR 2.0.50727; Media Center PC 6.0)",
        "Mozilla/5.0 (compatible; MSIE 8.0; Windows NT 6.0; Trident/4.0; WOW64; Trident/4.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; .NET CLR 1.0.3705; .NET CLR 1.1.4322)",
        "Mozilla/4.0 (compatible; MSIE 7.0b; Windows NT 5.2; .NET CLR 1.1.4322; .NET CLR 2.0.50727; InfoPath.2; .NET CLR 3.0.04506.30)",
        "Mozilla/5.0 (Windows; U; Windows NT 5.1; zh-CN) AppleWebKit/523.15 (KHTML, like Gecko, Safari/419.3) Arora/0.3 (Change: 287 c9dfb30)",
        "Mozilla/5.0 (X11; U; Linux; en-US) AppleWebKit/527+ (KHTML, like Gecko, Safari/419.3) Arora/0.6",
        "Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.1.2pre) Gecko/20070215 K-Ninja/2.1.1",
        "Mozilla/5.0 (Windows; U; Windows NT 5.1; zh-CN; rv:1.9) Gecko/20080705 Firefox/3.0 Kapiko/3.0",
        "Mozilla/5.0 (X11; Linux i686; U;) Gecko/20070322 Kazehakase/0.4.5",
        "Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.0.8) Gecko Fedora/1.9.0.8-1.fc10 Kazehakase/0.5.6",
        "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/535.11 (KHTML, like Gecko) Chrome/17.0.963.56 Safari/535.11",
        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_7_3) AppleWebKit/535.20 (KHTML, like Gecko) Chrome/19.0.1036.7 Safari/535.20",
        "Opera/9.80 (Macintosh; Intel Mac OS X 10.6.8; U; fr) Presto/2.9.168 Version/11.52",
        "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/536.11 (KHTML, like Gecko) Chrome/20.0.1132.11 TaoBrowser/2.0 Safari/536.11",
        "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.1 (KHTML, like Gecko) Chrome/21.0.1180.71 Safari/537.1 LBBROWSER",
        "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; WOW64; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E; LBBROWSER)",
        "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; QQDownload 732; .NET4.0C; .NET4.0E; LBBROWSER)",
        "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/535.11 (KHTML, like Gecko) Chrome/17.0.963.84 Safari/535.11 LBBROWSER",
        "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E)",
        "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; WOW64; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E; QQBrowser/7.0.3698.400)",
        "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; QQDownload 732; .NET4.0C; .NET4.0E)",
        "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1; Trident/4.0; SV1; QQDownload 732; .NET4.0C; .NET4.0E; 360SE)",
        "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; QQDownload 732; .NET4.0C; .NET4.0E)",
        "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E)",
        "Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.1 (KHTML, like Gecko) Chrome/21.0.1180.89 Safari/537.1",
        "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.1 (KHTML, like Gecko) Chrome/21.0.1180.89 Safari/537.1",
        "Mozilla/5.0 (iPad; U; CPU OS 4_2_1 like Mac OS X; zh-cn) AppleWebKit/533.17.9 (KHTML, like Gecko) Version/5.0.2 Mobile/8C148 Safari/6533.18.5",
        "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:2.0b13pre) Gecko/20110307 Firefox/4.0b13pre",
        "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:16.0) Gecko/20100101 Firefox/16.0",
        "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.64 Safari/537.11",
        "Mozilla/5.0 (X11; U; Linux x86_64; zh-CN; rv:1.9.2.10) Gecko/20100922 Ubuntu/10.10 (maverick) Firefox/3.6.10",
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36",
        ]
    USER_AGENT = random.choice(USER_AGENT_LIST)
    headers = {'user-agent': USER_AGENT}
    s = requests.Session()
    req = s.get(url, headers = headers)
    soup = BeautifulSoup(req.text, "html.parser")
    return soup

def csv_column_104(path_csv, key_txt): #建立行標題
    with open(path_csv + '.csv', mode='a+', newline='', encoding='utf-8') as employee_file:
        employee_writer = csv.writer(employee_file, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
        employee_writer.writerow(['日期', '工作名稱', '公司名稱', '公司地址', '薪資', '工作內容', '地區', '經歷', '學歷', '公司人數', '文章編號', '工作網址'])

def find_title_104(key_txt):
    #路徑組合
    today = get_todate()
    path_csv = "%s" % os.getcwd() + '/' + 'jobs_csv/'+ str(today) + key_txt + '_104人力銀行'
    if not os.path.isdir('jobs_csv'): # 確認是否有jobs_csv資料夾  沒有則返回Ture
        os.mkdir('jobs_csv') # 建立jobs_csv資料夾
        print('建立jobs_csv資料夾完成')
    csv_column_104(path_csv, key_txt) #建立行標題
    csv_save = ""
    key = quote(key_txt)
    #  104 api searchTempExclude=2  -> 設定排除派遣
    find_page_url = 'https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword={0}&order=15&asc=0&page=1&mode=s&jobsource=2018indexpoc&searchTempExclude=2'.format(key)
    get_sum_page = int(selenium_get_Code_104(find_page_url))
    print('共有：' + str(get_sum_page) + ' 頁')
    for i in tqdm(range(1, get_sum_page+1)):  #set page 1 to find all max page ,tqdm讀取進度條
        url = 'https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword={0}&order=15&asc=0&page={1}&mode=s&jobsource=2018indexpoc&searchTempExclude=2'.format(key, i)
        #time.sleep(random.randint(2,10)) #隨機等待
        soup = read_url(url) #讀取網頁
        print('目前爬取頁面是：' + url)
        for title_1 in soup.select('.b-block__left'):
            #有三個資料是無資料的，遇到無資料就跳過這個迴圈
            if title_1.select('.b-list-inline.b-clearfix.job-list-item__company') != soup.select('.b-block__left')[0].select('.b-list-inline.b-clearfix.job-list-item__company'):
                #日期
                try:
                    #正常代表找到 讚 廣告 (業主買廣告)，發生異常代表找不到 讚，執行except找日期
                    date_match__ = title_1.select('.b-icon--gray.b-icon--w18')[0].select('use')[0]
                    date = '廣告'
                except:
                    date = title_1.select('.b-tit__date')[0].get_text().replace('\n','').replace(' ','')

                #地區
                area = title_1.select('.b-list-inline.b-clearfix.job-list-intro.b-content')[0].find('li').get_text()
                #經歷(年資)
                experience = title_1.select('.b-list-inline.b-clearfix.job-list-intro.b-content')[0].find_all('li')[1].get_text()
                try: #業者沒有輸入學歷，遇到錯誤處理
                    #學歷
                    education = title_1.select('.b-list-inline.b-clearfix.job-list-intro.b-content')[0].find_all('li')[2].get_text()
                except:
                    education = ""
                #工作網址
                title_url = title_1.select('.js-job-link')[0].get('href')[2:]
                #get 文章編號
                title_str = title_url.split('?')[0].split('/')[-1] #get 文章編號
                #標題名稱
                title = title_1.select('.js-job-link')[0].get_text() #get title
                #print(title + title_url + area)
                #公司名
                company_name = title_1.select('li')[1].find('a').get('title').split('\n')[0][4:]
                try:
                    #公司地址
                    company_address = title_1.select('li')[1].find('a').get('title').split('\n')[1][5:]
                except:
                    company_address = ""
                try:
                    #簡介
                    introduction = title_1.select('.job-list-item__info.b-clearfix.b-content')[0].get_text()
                    #處理string \r \n5 \n轉成''
                    introduction = introduction.replace('\r','').replace('\n5','').replace('\n','')
                except:
                    introduction = ""
                #薪資
                try:
                    salary = title_1.select('.b-tag--default')[0].get_text()
                except:
                    salary = 0 #沒有寫薪資或待遇面議，設定 0

                if salary == '待遇面議':
                    salary = "待遇面議"
                else: #數字處理 25000~35000 取25000最低薪資為主要，三位數 = 日薪，四位數 = 論件計酬
                    try:
                        salary = re.search('\d+.\d+', salary).group()
                    except:
                        salary = 0
                #員工人數
                try:
                    people = title_1.select('.b-tag--default')[1].get_text()
                except:
                    people = ""
                #clear_output() # 清除輸出 用於清除進度讀，註解#不使用：用來檢查出錯的網址

                with open(path_csv + '.csv', mode='a+', newline='', encoding='utf-8') as employee_file: #w
                    employee_writer = csv.writer(employee_file, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
                    employee_writer.writerow([date, title, company_name, company_address, salary, introduction, area, experience, education, people, title_str, title_url])
            else:
                continue
    return print('爬取104完成：請開啟csv檔案')


# input_go = input('輸入關鍵字')
# save_title_data = find_title_104(input_go)




############ 1111人力銀行 #########


def selenium_get_Code_1111(url):
    chrome_options = webdriver.ChromeOptions()
    chrome_options.add_argument('--headless')
    chrome_options.add_argument('--no-sandbox')
    chrome_options.add_argument('--disable-dev-shm-usage')
    driver = webdriver.Chrome('chromedriver',options=chrome_options)
    driver.get(url)
    save = driver.page_source
    driver.quit()#關閉瀏覽器
    soup = BeautifulSoup(save, "html.parser")

    page = soup.select('.custom-select')[0].select('option')[0].text
    page = page.split('/')
    page = page[1].strip(' ')
    return page


def csv_column_1111(path_csv): #建立行標題
    with open(path_csv + '.csv', mode='a+', newline='', encoding='utf-8') as employee_file:
        employee_writer = csv.writer(employee_file, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
        employee_writer.writerow(['日期', '工作名稱', '公司名稱', '公司地址', '薪資', '工作內容', '地區', '經歷', '學歷', '工作網址'])

def find_data_1111(soup):
    #錢錢
    mnone = soup.select('.needs')
    #縣市區域
    location = soup.select('.needs')
    #日期
    get_date = soup.select('.date')
    #簡介
    jbInfoTxt = soup.select('.jbInfoTxt')
    #網址
    jobs_url = soup.select('.position0')
    #公司名稱、類別、住址
    company_data = soup.select('.d-md-flex')
    #工作標題
    title = soup.select('.position0')
    #工作經驗
    jobs_exp = soup.select('.needs')
    # 學歷
    education = soup.select('.needs')
    return mnone, location, jbInfoTxt, jobs_url, company_data, title, jobs_exp, get_date, education



def find_title_1111(key_txt):
    #路徑組合
    today = get_todate()
    path_csv = 'jobs_csv/' + str(today) + key_txt + '_1111人力銀行'
    if not os.path.isdir('jobs_csv'): # 確認是否有jobs_csv資料夾  沒有則返回Ture
        os.mkdir('jobs_csv') # 建立jobs_csv資料夾
        print('建立jobs_csv資料夾完成')
    csv_column_1111(path_csv) #建立行標題
    key = quote(key_txt)
    find_page_url = 'https://www.1111.com.tw/job-bank/job-index.asp?si=1&ss=s&ks={0}&page=1'.format(key)
    #取得最大page數
    get_sum_page = int(selenium_get_Code_1111(find_page_url))
    print('共有：' + str(get_sum_page) + ' 頁')

    for i in tqdm(range(1, get_sum_page+1)):
        url = 'https://www.1111.com.tw/job-bank/job-index.asp?si=1&ss=s&ks={0}&page={1}'.format(key, i)
        soup = read_url(url) #讀取網頁
        #讀取網頁資料
        mnone, location, jbInfoTxt, jobs_url, company_data, title, jobs_exp, get_date, education = find_data_1111(soup)
        print('目前爬取頁面是：' + url)
        for mnone, location, jbInfoTxt, jobs_url, company_data, title, jobs_exp, get_date, education in zip(mnone, location, jbInfoTxt, jobs_url, company_data, title, jobs_exp, get_date, education):
            #錢 取最低薪資
            try:
                mnone = mnone.find_all("span")[1].get_text()
                get_mone = re.search('\d+.\d+', mnone).group()
            except:
                get_mone = '面議（經常性薪資4萬/月含以上）' #也可以直接給40000
            #日期
            get_date = get_date.get_text()[5:]

            #縣市區域
            location = location.find_all("span")[0].get_text()

            #簡介
            jbInfoTxt = jbInfoTxt.get_text().replace("\xa0", "") #刪除\xa0

            #工作網址
            jobs_url = 'https://www.1111.com.tw{0}'.format(jobs_url.find('a').get('href'))

            #公司名
            company = company_data.find_all('a')[0].get('title').replace('\r','').split('\n')[0][6:]

            #公司分類 目前暫不使用
            category = company_data.find_all('a')[0].get('title').replace('\r','').split('\n')[1][6:]

            #公司地址
            address = company_data.find_all('a')[0].get('title').replace('\r','').split('\n')[2][6:]

            #工作標題
            title = title.find('a').get('title')

            # 工作經驗
            jobs_exp = jobs_exp.find_all("span")[2].get_text()

            # 學歷
            education = education.find_all("span")[3].get_text()
            # 儲存

            # clear_output() # 清除輸出 用於清除進度讀

            with open(path_csv + '.csv', mode='a+', newline='', encoding='utf-8') as employee_file: #w
                employee_writer = csv.writer(employee_file, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
                employee_writer.writerow([get_date, title, company, address, get_mone, jbInfoTxt, location, jobs_exp, education, jobs_url])
    return print('爬取1111完成：請開啟csv檔案')

In [ ]:
import os
os.getcwd()
os.chdir('/content/drive/MyDrive/colab_could/104')

In [ ]:
sel = input('請選擇要使用104或是1111，兩種都要爬輸入 2：')

while True:
    if sel == '104':
        input_str = input('請輸入要搜尋的關鍵字，取消請輸入0：')
        if input_str == '0':
            print('已經取消')
            break
        find_title_104(input_str)
        break
    elif sel == '1111':
        input_str = input('請輸入要搜尋的關鍵字，取消請輸入0：')
        if input_str == '0':
            print('已經取消')
            break
        find_title_1111(input_str)
        break
    elif sel == '0':
        break
    elif sel == '2':
        input_str = input('請輸入要搜尋的關鍵字，取消請輸入0：')
        if input_str == '0':
            print('已經取消')
            break
        print('爬取104、1111')
        find_title_104(input_str)
        find_title_1111(input_str)
        break

    else:
        print('請輸入正確的選項：輸入 104 或 1111：')
        sel = input('請選擇要使用104或是1111：')

請選擇要使用104或是1111，兩種都要爬輸入 2：2
請輸入要搜尋的關鍵字，取消請輸入0：data analysis、business analysis
爬取104、1111
建立jobs_csv資料夾完成
共有：150 頁


  0%|          | 0/150 [00:00<?, ?it/s]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=1&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  1%|▏         | 2/150 [00:03<04:13,  1.71s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=2&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  2%|▏         | 3/150 [00:05<03:58,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=3&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  3%|▎         | 4/150 [00:06<04:10,  1.72s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=4&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  3%|▎         | 5/150 [00:08<04:08,  1.72s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=5&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  4%|▍         | 6/150 [00:10<04:02,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=6&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  5%|▍         | 7/150 [00:11<03:59,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=7&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  5%|▌         | 8/150 [00:13<03:59,  1.69s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=8&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  6%|▌         | 9/150 [00:15<03:48,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=9&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  7%|▋         | 10/150 [00:16<03:48,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=10&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  7%|▋         | 11/150 [00:18<03:44,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=11&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  8%|▊         | 12/150 [00:19<03:40,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=12&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  9%|▊         | 13/150 [00:21<03:35,  1.57s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=13&mode=s&jobsource=2018indexpoc&searchTempExclude=2


  9%|▉         | 14/150 [00:23<03:36,  1.59s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=14&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 10%|█         | 15/150 [00:24<03:37,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=15&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 11%|█         | 16/150 [00:26<03:41,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=16&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 11%|█▏        | 17/150 [00:28<03:38,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=17&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 12%|█▏        | 18/150 [00:29<03:31,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=18&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 13%|█▎        | 19/150 [00:31<03:32,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=19&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 13%|█▎        | 20/150 [00:32<03:27,  1.59s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=20&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 14%|█▍        | 21/150 [00:34<03:24,  1.58s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=21&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 15%|█▍        | 22/150 [00:35<03:22,  1.58s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=22&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 15%|█▌        | 23/150 [00:37<03:33,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=23&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 16%|█▌        | 24/150 [00:39<03:27,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=24&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 17%|█▋        | 25/150 [00:41<03:27,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=25&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 17%|█▋        | 26/150 [00:42<03:24,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=26&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 18%|█▊        | 27/150 [00:44<03:20,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=27&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 19%|█▊        | 28/150 [00:45<03:21,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=28&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 19%|█▉        | 29/150 [00:47<03:22,  1.67s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=29&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 20%|██        | 30/150 [00:49<03:16,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=30&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 21%|██        | 31/150 [00:50<03:10,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=31&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 21%|██▏       | 32/150 [00:52<03:05,  1.58s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=32&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 22%|██▏       | 33/150 [00:53<03:07,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=33&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 23%|██▎       | 34/150 [00:55<03:03,  1.58s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=34&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 23%|██▎       | 35/150 [00:56<02:59,  1.56s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=35&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 24%|██▍       | 36/150 [00:58<03:02,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=36&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 25%|██▍       | 37/150 [01:00<03:05,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=37&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 25%|██▌       | 38/150 [01:01<03:00,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=38&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 26%|██▌       | 39/150 [01:03<02:59,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=39&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 27%|██▋       | 40/150 [01:05<02:58,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=40&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 27%|██▋       | 41/150 [01:06<02:56,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=41&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 28%|██▊       | 42/150 [01:08<02:59,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=42&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 29%|██▊       | 43/150 [01:10<02:56,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=43&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 29%|██▉       | 44/150 [01:11<02:53,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=44&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 30%|███       | 45/150 [01:14<03:19,  1.90s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=45&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 31%|███       | 46/150 [01:15<03:05,  1.79s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=46&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 31%|███▏      | 47/150 [01:17<02:57,  1.72s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=47&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 32%|███▏      | 48/150 [01:19<02:59,  1.76s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=48&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 33%|███▎      | 49/150 [01:20<02:52,  1.71s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=49&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 33%|███▎      | 50/150 [01:22<03:02,  1.82s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=50&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 34%|███▍      | 51/150 [01:24<02:56,  1.78s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=51&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 35%|███▍      | 52/150 [01:26<02:52,  1.76s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=52&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 35%|███▌      | 53/150 [01:27<02:45,  1.71s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=53&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 36%|███▌      | 54/150 [01:29<02:43,  1.70s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=54&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 37%|███▋      | 55/150 [01:31<02:36,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=55&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 37%|███▋      | 56/150 [01:33<02:59,  1.91s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=56&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 38%|███▊      | 57/150 [01:35<02:46,  1.79s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=57&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 39%|███▊      | 58/150 [01:36<02:36,  1.70s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=58&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 39%|███▉      | 59/150 [01:38<02:28,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=59&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 40%|████      | 60/150 [01:39<02:30,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=60&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 41%|████      | 61/150 [01:41<02:29,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=61&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 41%|████▏     | 62/150 [01:43<02:23,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=62&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 42%|████▏     | 63/150 [01:45<02:33,  1.77s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=63&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 43%|████▎     | 64/150 [01:46<02:26,  1.71s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=64&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 43%|████▎     | 65/150 [01:48<02:18,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=65&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 44%|████▍     | 66/150 [01:49<02:17,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=66&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 45%|████▍     | 67/150 [01:51<02:14,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=67&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 45%|████▌     | 68/150 [01:53<02:18,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=68&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 46%|████▌     | 69/150 [01:54<02:14,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=69&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 47%|████▋     | 70/150 [01:56<02:12,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=70&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 47%|████▋     | 71/150 [01:58<02:09,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=71&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 48%|████▊     | 72/150 [01:59<02:12,  1.70s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=72&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 49%|████▊     | 73/150 [02:01<02:06,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=73&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 49%|████▉     | 74/150 [02:03<02:06,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=74&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 50%|█████     | 75/150 [02:04<02:01,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=75&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 51%|█████     | 76/150 [02:06<01:58,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=76&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 51%|█████▏    | 77/150 [02:07<01:57,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=77&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 52%|█████▏    | 78/150 [02:09<01:58,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=78&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 53%|█████▎    | 79/150 [02:11<01:54,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=79&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 53%|█████▎    | 80/150 [02:12<01:51,  1.59s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=80&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 54%|█████▍    | 81/150 [02:14<01:50,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=81&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 55%|█████▍    | 82/150 [02:15<01:47,  1.59s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=82&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 55%|█████▌    | 83/150 [02:17<01:45,  1.58s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=83&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 56%|█████▌    | 84/150 [02:19<01:46,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=84&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 57%|█████▋    | 85/150 [02:20<01:46,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=85&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 57%|█████▋    | 86/150 [02:22<01:44,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=86&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 58%|█████▊    | 87/150 [02:24<01:43,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=87&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 59%|█████▊    | 88/150 [02:25<01:42,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=88&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 59%|█████▉    | 89/150 [02:27<01:41,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=89&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 60%|██████    | 90/150 [02:28<01:37,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=90&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 61%|██████    | 91/150 [02:30<01:34,  1.59s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=91&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 61%|██████▏   | 92/150 [02:32<01:35,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=92&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 62%|██████▏   | 93/150 [02:33<01:32,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=93&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 63%|██████▎   | 94/150 [02:35<01:30,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=94&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 63%|██████▎   | 95/150 [02:37<01:36,  1.76s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=95&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 64%|██████▍   | 96/150 [02:39<01:32,  1.72s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=96&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 65%|██████▍   | 97/150 [02:40<01:30,  1.70s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=97&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 65%|██████▌   | 98/150 [02:42<01:27,  1.69s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=98&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 66%|██████▌   | 99/150 [02:44<01:25,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=99&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 67%|██████▋   | 100/150 [02:45<01:22,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=100&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 67%|██████▋   | 101/150 [02:47<01:18,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=101&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 68%|██████▊   | 102/150 [02:48<01:16,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=102&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 69%|██████▊   | 103/150 [02:50<01:17,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=103&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 69%|██████▉   | 104/150 [02:52<01:16,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=104&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 70%|███████   | 105/150 [02:53<01:12,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=105&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 71%|███████   | 106/150 [02:55<01:10,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=106&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 71%|███████▏  | 107/150 [02:58<01:23,  1.93s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=107&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 72%|███████▏  | 108/150 [02:59<01:16,  1.81s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=108&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 73%|███████▎  | 109/150 [03:01<01:11,  1.74s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=109&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 73%|███████▎  | 110/150 [03:02<01:08,  1.72s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=110&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 74%|███████▍  | 111/150 [03:04<01:07,  1.74s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=111&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 75%|███████▍  | 112/150 [03:06<01:05,  1.72s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=112&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 75%|███████▌  | 113/150 [03:07<01:01,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=113&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 76%|███████▌  | 114/150 [03:09<00:58,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=114&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 77%|███████▋  | 115/150 [03:10<00:55,  1.59s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=115&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 77%|███████▋  | 116/150 [03:12<00:55,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=116&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 78%|███████▊  | 117/150 [03:14<00:53,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=117&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 79%|███████▊  | 118/150 [03:16<00:56,  1.76s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=118&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 79%|███████▉  | 119/150 [03:17<00:53,  1.73s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=119&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 80%|████████  | 120/150 [03:19<00:50,  1.69s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=120&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 81%|████████  | 121/150 [03:21<00:47,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=121&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 81%|████████▏ | 122/150 [03:22<00:47,  1.71s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=122&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 82%|████████▏ | 123/150 [03:24<00:44,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=123&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 83%|████████▎ | 124/150 [03:25<00:41,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=124&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 83%|████████▎ | 125/150 [03:27<00:41,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=125&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 84%|████████▍ | 126/150 [03:29<00:39,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=126&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 85%|████████▍ | 127/150 [03:30<00:36,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=127&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 85%|████████▌ | 128/150 [03:32<00:36,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=128&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 86%|████████▌ | 129/150 [03:34<00:34,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=129&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 87%|████████▋ | 130/150 [03:35<00:33,  1.67s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=130&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 87%|████████▋ | 131/150 [03:37<00:31,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=131&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 88%|████████▊ | 132/150 [03:39<00:28,  1.60s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=132&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 89%|████████▊ | 133/150 [03:40<00:27,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=133&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 89%|████████▉ | 134/150 [03:42<00:26,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=134&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 90%|█████████ | 135/150 [03:44<00:24,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=135&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 91%|█████████ | 136/150 [03:45<00:23,  1.65s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=136&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 91%|█████████▏| 137/150 [03:47<00:21,  1.69s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=137&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 92%|█████████▏| 138/150 [03:49<00:19,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=138&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 93%|█████████▎| 139/150 [03:50<00:17,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=139&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 93%|█████████▎| 140/150 [03:52<00:16,  1.62s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=140&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 94%|█████████▍| 141/150 [03:54<00:15,  1.75s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=141&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 95%|█████████▍| 142/150 [03:55<00:13,  1.69s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=142&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 95%|█████████▌| 143/150 [03:57<00:11,  1.69s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=143&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 96%|█████████▌| 144/150 [03:59<00:10,  1.67s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=144&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 97%|█████████▋| 145/150 [04:00<00:08,  1.63s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=145&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 97%|█████████▋| 146/150 [04:02<00:06,  1.68s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=146&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 98%|█████████▊| 147/150 [04:04<00:04,  1.64s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=147&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 99%|█████████▊| 148/150 [04:05<00:03,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=148&mode=s&jobsource=2018indexpoc&searchTempExclude=2


 99%|█████████▉| 149/150 [04:07<00:01,  1.61s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=149&mode=s&jobsource=2018indexpoc&searchTempExclude=2


100%|██████████| 150/150 [04:08<00:00,  1.66s/it]

目前爬取頁面是：https://www.104.com.tw/jobs/search/?ro=0&kwop=7&keyword=data%20analysis%E3%80%81business%20analysis&order=15&asc=0&page=150&mode=s&jobsource=2018indexpoc&searchTempExclude=2
爬取104完成：請開啟csv檔案


IndexError: ignored

###👉Pandas資料分析 ###
* ##### 1.讀取資料 #####   

In [ ]:
import glob
glob.glob('jobs_csv/*.csv')

['jobs_csv/2022-11-27data analysis、business analysis_104人力銀行.csv',
 'jobs_csv/2022-11-27data analysis、business analysis_1111人力銀行.csv']

In [ ]:
#讀取爬取到的資料 (需修改檔名)
data_104 = pd.read_csv('jobs_csv/2022-11-27data analysis、business analysis_104人力銀行.csv')
data_1111 = pd.read_csv('jobs_csv/2022-11-27data analysis、business analysis_1111人力銀行.csv')

* ##### 2.查看資料 #####

In [ ]:
#查看資料
data_104.head()

日期                                               工作名稱               公司名稱  \
0  廣告                                 Accounting Analyst           奇方動力有限公司   
1  廣告                                   BJ2 資深系統工程師/技術主任       緯穎科技服務股份有限公司   
2  廣告         BEZ3-Signal Integrity Simulation工程師 (無經驗可)       緯穎科技服務股份有限公司   
3  廣告  Business Development Sr. Coordinator/Superviso...        美商賀寶芙股份有限公司   
4  廣告         Senior Executive, Finance & Administration  香港商香港旅發局有限公司台灣分公司   

                 公司地址      薪資  \
0      台北市信義區松德路65號4樓    待遇面議   
1   新北市汐止區新台五路一段90號8樓  30,000   
2   新北市汐止區新台五路一段90號8樓    待遇面議   
3   台北市南港區市民大道七段8號16樓    待遇面議   
4  台北市中正區忠孝西路一段66號29樓    待遇面議   

                                                工作內容      地區    經歷  學歷   公司人數  \
0  We are an established e-commerce company headq...  台北市信義區  1年以上  專科  員工32人   
1  【工作內容】1. 辦公室系統管理及維護（Microsoft AD、KMS、DNS、WSUS、...  新北市汐止區  3年以上  大學   上市上櫃   
2  In this position, you will be responsible for ...  新北市汐止區  經歷不拘  大學   上市上櫃   
3  Position Summary: As TW Herbalife communicatio...  台北市南港區  5年以上  大學   外商公司   
4  Responsibilities• To keep abreast of current l...  台北市中正區  3年以上  大學   員工9人   

    文章編號                                           工作網址  
0  7c6oz  www.104.com.tw/job/7c6oz?jobsource=hotjob_chr  
1  7ltmv  www.104.com.tw/job/7ltmv?jobsource=hotjob_chr  
2  3v28w  www.104.com.tw/job/3v28w?jobsource=hotjob_chr  
3  7r76s  www.104.com.tw/job/7r76s?jobsource=hotjob_chr  
4  3cipt  www.104.com.tw/job/3cipt?jobsource=hotjob_chr

In [ ]:
data_1111.head()

Empty DataFrame
Columns: [日期, 工作名稱, 公司名稱, 公司地址, 薪資, 工作內容, 地區, 經歷, 學歷, 工作網址]
Index: []

In [ ]:
# 檢查有多少筆資料 共2870筆
data_104.shape

(2870, 12)

* #### 3.選擇想要的資料 ####
日期、工作名稱、公司地址、薪資、工作內容、地區、經歷、學歷、工作網址 依想要的順序排序

In [ ]:
new_data_1111 =  data_1111[['日期', '工作名稱', '薪資', '地區', '工作內容', '公司地址', '經歷', '學歷', '工作網址']]
new_data_104 = data_104[['日期', '工作名稱', '薪資', '地區', '工作內容', '公司地址', '經歷', '學歷', '工作網址']]

In [ ]:
#檢查shape
print('104：' + str(new_data_104.shape))
#print('1111：' + str(new_data_1111.shape))

104：(2870, 9)
1111：(0, 9)


In [ ]:
new_data_104.tail() #資料最後面的index 不是4733，需要重新建立index 不包含標題 所以要減去一筆資料 是4733

日期                                               工作名稱      薪資  \
2865  11/11            Project Manager (Supply Chain Planning)    待遇面議   
2866  11/25                              【新竹】RF工程師 RF Engineer    待遇面議   
2867  11/23  Software Engineer - BSP (Board Support Package...    待遇面議   
2868  11/22                                           品管人員(石岡)  33,000   
2869  11/23                      Senior SIPI Engineer _ PLL 台南    待遇面議   

          地區                                               工作內容  \
2865     新竹市   risk management, inventory optimization, on-t...   
2866  新竹縣竹北市   PCB layout review (EMI / EMC / noise preventi...   
2867  台北市大同區  Job Title: Value Engineer Report to: Director ...   
2868  台中市石岡區  1.廠內混凝土品質測試。2.溝通協調廠內外製程混凝土品質管制，協助解決工地品質變異狀況，控制...   
2869  台南市歸仁區   is high-speed IC package design/analysis. As ...   

                   公司地址    經歷  學歷  \
2865        新竹市金山八街1號2樓  經歷不拘  大學   
2866     新竹縣竹北市復興二路251號  5年以上  大學   
2867  台北市大同區承德路１段40號10樓  5年以上  大學   
2868     台中市石岡區豐勢路1巷60號  經歷不拘  高中   
2869             台南市歸仁區  5年以上  大學   

                                                   工作網址  
2865  www.104.com.tw/job/7tcas?jobsource=jolist_b_re...  
2866  www.104.com.tw/job/7qmue?jobsource=jolist_b_re...  
2867  www.104.com.tw/job/7t3k6?jobsource=jolist_b_re...  
2868  www.104.com.tw/job/6os4q?jobsource=jolist_b_re...  
2869  www.104.com.tw/job/7tj2f?jobsource=jolist_b_re...

In [ ]:
new_data_104.reset_index(drop=True, inplace=True) #drop=True 刪除原有index inplace 直接更新現有的DataFrame

In [ ]:
new_data_104.tail()

日期                                               工作名稱      薪資  \
2865  11/11            Project Manager (Supply Chain Planning)    待遇面議   
2866  11/25                              【新竹】RF工程師 RF Engineer    待遇面議   
2867  11/23  Software Engineer - BSP (Board Support Package...    待遇面議   
2868  11/22                                           品管人員(石岡)  33,000   
2869  11/23                      Senior SIPI Engineer _ PLL 台南    待遇面議   

          地區                                               工作內容  \
2865     新竹市   risk management, inventory optimization, on-t...   
2866  新竹縣竹北市   PCB layout review (EMI / EMC / noise preventi...   
2867  台北市大同區  Job Title: Value Engineer Report to: Director ...   
2868  台中市石岡區  1.廠內混凝土品質測試。2.溝通協調廠內外製程混凝土品質管制，協助解決工地品質變異狀況，控制...   
2869  台南市歸仁區   is high-speed IC package design/analysis. As ...   

                   公司地址    經歷  學歷  \
2865        新竹市金山八街1號2樓  經歷不拘  大學   
2866     新竹縣竹北市復興二路251號  5年以上  大學   
2867  台北市大同區承德路１段40號10樓  5年以上  大學   
2868     台中市石岡區豐勢路1巷60號  經歷不拘  高中   
2869             台南市歸仁區  5年以上  大學   

                                                   工作網址  
2865  www.104.com.tw/job/7tcas?jobsource=jolist_b_re...  
2866  www.104.com.tw/job/7qmue?jobsource=jolist_b_re...  
2867  www.104.com.tw/job/7t3k6?jobsource=jolist_b_re...  
2868  www.104.com.tw/job/6os4q?jobsource=jolist_b_re...  
2869  www.104.com.tw/job/7tj2f?jobsource=jolist_b_re...

* #### 清理資料 ####
>ex:清理 薪資的資料：
>1. 將 待遇面議 轉換成 0
>2. 將 面議（經常性薪資4萬/月含以上）轉成 40000
>3. 將 時薪 例如：150~200 轉成只取最低時薪 150
>4. 移除數字中的, 例如"30,000"將,移除

In [ ]:
#將四萬改成0  但是也可以改成40,000
new_data_104['薪資'] = new_data_104.薪資.str.replace('面議（經常性薪資4萬/月含以上）', '40000')

#將待遇面議改成0
new_data_104['薪資'] = new_data_104.薪資.str.replace('待遇面議', '0')

#時薪 取最低時薪
new_data_104.loc[new_data_104.薪資.str.contains('~'),'薪資'] = new_data_104.loc[new_data_104.薪資.str.contains('~'),'薪資'].str.split('~').str[0]

#數字有"30,000"將,移除
new_data_104.loc[:,'薪資'] = new_data_104['薪資'].apply(lambda x:x.replace(',',''))

# 薪資轉成int type
# concat_104_1111.loc[:,'薪資'] = concat_104_1111['薪資'].astype(int)
new_data_104.loc[:,'薪資'] = pd.to_numeric(new_data_104['薪資'])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """
/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-v

In [ ]:
#薪資 已轉成 int64
new_data_104.dtypes

日期      object
工作名稱    object
薪資       int64
地區      object
工作內容    object
公司地址    object
經歷      object
學歷      object
工作網址    object
dtype: object

* #### 過濾資料 ####
>找想要的工作，
>過濾方法：可以參考這篇 https://ithelp.ithome.com.tw/articles/10194003

In [ ]:
#1萬至3萬之間、地區：台北市 這種過濾方法會寫越寫長
#concat_104_1111.loc[concat_104_1111.薪資 > 10000].loc[concat_104_1111.薪資 < 30000].loc[concat_104_1111.地區.str.contains('台北市')]


#4萬以上 找雙北 關鍵字：分析
mask1 = new_data_104.薪資 >= 50000
mask2 = new_data_104.地區.str.contains('台北市')
mask3 = new_data_104.地區.str.contains('新北市')
mask4 = new_data_104.地區.str.contains('台中市')
mask5 = new_data_104.工作名稱.str.contains('分析')
mask6 = new_data_104.工作名稱.str.contains('data')
mask7 = new_data_104.工作名稱.str.contains('business')
mask8 = new_data_104.工作內容.str.contains('金融') #工作內容簡介也許會有相關的工作
mask9 = new_data_104.工作內容.str.contains('分析')
mask10 = new_data_104.工作內容.str.contains('股票')



#搜尋 3萬以上 工作名稱有"證卷"關鍵字、區域 台北市 新北市
# & = and , | = or
new_data_104.loc[((mask2 | mask3 | mask4) & mask1 & mask5) | ((mask2 | mask3 | mask4) & mask1 & mask10)]


日期                                               工作名稱      薪資  \
50    11/23                         【採線上面談】 Data Analyst 資料分析師   50000   
104   11/22            [BA] 軟體商業分析師 IT Business Analyst / 線上面談   50000   
160   11/24                      資深資料分析師 (Senior Data Analyst)   55000   
179   11/23                 【採線上面談】Senior Data Analyst 資深資料分析師   90000   
191   11/25                               Data Analyst / 資料分析師   57750   
269   11/21                                 Data Analyst 數據分析師   50000   
270   11/21                                 Data Analyst 數據分析師   50000   
282   11/04                           Business Analyst 商業流程分析師   70000   
296   11/23                           Business Analyst / 商業分析師  600000   
315   11/24  經營分析主任管理師<台北總部>(Management  analysis_ Principa...   60000   
349   11/16                             Business Analyst 商業分析師   60000   
528   11/23                          (Sr.) BI Analyst  BI 數據分析  650000   
701   11/24                           Data engineer/數據工程師 (分析)   60000   
764   11/22                                              系統分析師   60000   
906   11/21                              Data Engineer 數據分析工程師  700000   
1257  11/21                                              數據分析師   50000   
1258  11/24                 Environmental Data Analyst 環境資料分析師  500000   
1379  11/22            關鍵字付費廣告行銷人員 (遠端)｜廣告投放、關鍵字分析、數據分析｜外商科技新創  480000   
1913  11/23  【FMCG美商】誠徵 財務分析暨稅務主任 (工作型態穩定/ 不須負擔管理責任/ 歡迎熟悉稅務...  700000   
1930  11/23  【FMCG美商】誠徵 財務分析暨稅務主任 (工作型態穩定/ 不須負擔管理責任/ 歡迎熟悉稅務...  700000   
1973  11/24                                系統分析師 / SA Engineer   55000   

          地區                                               工作內容  \
50    台北市南港區   a Data Analyst does in Titansoft】  ．Conduct d...   
104   台北市內湖區  你是產品開發團隊與業務需求部門中間不可或缺的角色。邏輯清晰，具備能夠把業務部門的需求(Bus...   
160   台北市中山區   consistency of data2.Analyze business related...   
179   台北市南港區   a Data Analyst does in Titansoft】  ．Conduct d...   
191   台北市內湖區  •  Discover insights from a wide variety of da...   
269   台北市松山區  The successful candidate will turn data into i...   
270   台北市松山區  The successful candidate will turn data into i...   
282   台北市松山區  保險商業流程需求分析人員(BA – Business Analyst)1. 明確定義保單行政...   
296   台北市大同區  請至『 https://reurl.cc/ROOqLr 』進行投遞，我們會以此平台收到的履歷...   
315   台北市大安區  1.主掌SRD事業處經營管理報告，透過內外部營運及財務分析，提供事業處經營與決策資訊2.掌握...   
349   台中市北屯區  - 熟悉數據分析技巧，解析公司與產業的現況，包含可能遇到的問題、潛在的機會點- 具備完善的溝...   
528   台北市中正區  - Set up and execute extract, transformation a...   
701   台北市中正區   welcome proposals for new tooling stacks.*** ...   
764   台北市中山區  1.負責電腦資訊之傳輸，以及電腦軟體、資料庫、資料結構之維修與研發。2.分析設計網路程式及界...   
906   台北市信義區  工作職責：1. ETL處理：負責基於ElasticSearch Beats 、AWS Ste...   
1257  台北市南港區  【職務內容】1. 建立、分析現有客戶資料，並提出數據洞察與建議。 2. CRM Data 數...   
1258  新北市新店區  你是否對於保護環境具有滿腔的熱血，並且深信在這個時代可以透過軟體改變許多事情呢?!本公司是一...   
1379  台北市大安區       management, digital media buying, PR and ...   
1913  台北市內湖區  ._Prepare quarterly tax provision schedule_Pre...   
1930  台北市內湖區  ._Prepare quarterly tax provision schedule_Pre...   
1973  台北市大安區  1. Software development project management: (s...   

                                    公司地址    經歷    學歷  \
50    台北市南港區台北市南港區園區街3之1號11樓之1（南港軟體園區G棟）  經歷不拘  學歷不拘   
104               台北市內湖區瑞光路513巷22弄5號6樓之4  3年以上    大學   
160                    台北市中山區樂群三路126號14樓  3年以上    大學   
179   台北市南港區台北市南港區園區街3之1號11樓之1（南港軟體園區G棟）  5年以上  學歷不拘   
191                       台北市內湖區瑞湖街58號3樓  1年以上    大學   
269                 台北市松山區民生東路三段156號8樓之1  3年以上    大學   
270                 台北市松山區民生東路三段156號8樓之1  3年以上    大學   
282                      台北市松山區復興北路99號6樓  3年以上    專科   
296                            台北市大同區承德路  1年以上    大學   
315                               台北市大安區  5年以上    大學   
349                        台中市北屯區河北二街78號  經歷不拘    大學   
528                台北市中正區杭州南路一段15-1號 16F  2年以上    高中   
701         台北市中正區忠孝東路二段112號9樓(忠孝新生2號出口)  3年以上    大學   
764                      台北市中山區長春路326號2樓  3年以上  學歷不拘   
906                          台北市信義區基隆路一段  1年以上    專科   
1257     

In [ ]:
#儲存成 Excel格式檔
file_name = str(get_todate()) + '104職缺-數據分析' #檔案名稱
save_excel = new_data_104.loc[((mask2 | mask3 | mask4) & mask1 & mask5) | ((mask2 | mask3 | mask4) & mask1 & mask10)]
save_excel.to_excel('jobs_csv/{}.xlsx'.format(file_name), sheet_name='passengers', index=False)

In [ ]:
#我想兼差打工 50至1000之間
new_data_104.loc[new_data_104.薪資 > 50].loc[new_data_104.薪資 < 1000]

日期                                               工作名稱   薪資      地區  \
19    11/09  Intern - Data Analysis Intern (1 year contract...  170  台中市西屯區   
22    11/15  RegTech Development & Data Analysis Intern (6 ...  230  台北市信義區   
36    11/21                              [Intern] Data Analyst  168  台北市信義區   
54    11/24                Data Analyst Intern / Shopee Xpress  180  台北市信義區   
61    11/23                       Data Analysis Intern 數據分析實習生  180  台北市信義區   
...     ...                                                ...  ...     ...   
2760  11/17                       單車喜客OAKLEY 專櫃兼職人員-台中三井Outlet  170  台中市梧棲區   
2784  11/22                            增編｜外場服務人員｜培訓後可轉儲備幹部｜八田源  190  高雄市鼓山區   
2801  11/24                         POKE GO波奇走走 兼職人員～即將開幕台北信義店  170  台北市信義區   
2818  11/24                         POKE GO波奇走走 兼職人員～即將開幕台北信義店  170  台北市信義區   
2844  11/09  Intern - Software Test Automation Intern (1-ye...  170  新北市林口區   

                                                   工作內容  \
19      This position will be located in Taichung.  ...   
22     data into information and turn information in...   
36     payment, contract and invoice with Finance an...   
54    Job Description:1. Support daily data request ...   
61    工作職責：實現數據提取和轉換；及時、準確地提供數據支持為策略開發蒐集整理並清洗數據，評估類似...   
...                                                 ...   
2760  我們徵求：1.充滿熱情的您2.認真、積極、努力以及肯挑戰的您3.有相關服務業/百貨業/運動品...   
2784  ｜人力增編｜1、適合晚睡晚起不想朝九晚五的你2、需要外向愛笑、反應機靈的你，能夠照顧外場客人...   
2801  工作內容1. 負責協助客人點餐，製餐以及結帳等工作。2. 餐具清潔，備料。3. 整理工作與環...   
2818  工作內容1. 負責協助客人點餐，製餐以及結帳等工作。2. 餐具清潔，備料。3. 整理工作與環...   
2844   analysis and system design of test automation...   

                           公司地址    經歷    學歷  \
19                       台中市西屯區  經歷不拘    大學   
22    台北市信義區信義路五段7號30樓(台北101大樓)  經歷不拘    大學   
36                台北市信義區松仁路123號  經歷不拘    大學   
54                 台北市信義區菸廠路88號  1年以上    大學   
61                  台北市信義區基隆路1段  經歷不拘    大學   
...                         ...   ...   ...   
2760     台中市梧棲區台中市梧棲區臺灣大道十段168號  經歷不拘    高中   
2784              高雄市鼓山區青海路189號  1年以上    高中   
2801            台北市信義區吳興街（波奇走走)  經歷不拘  學歷不拘   
2818            台北市信義區吳興街（波奇走走)  經歷不拘  學歷不拘   
2844          新北市林口區公道五路三段一號十一樓  經歷不拘    碩士   

                                                   工作網址  
19    www.104.com.tw/job/7st5l?jobsource=jolist_a_re...  
22    www.104.com.tw/job/6srxr?jobsource=jolist_a_re...  
36    www.104.com.tw/job/7svwu?jobsource=jolist_c_re...  
54    www.104.com.tw/job/7dnje?jobsource=jolist_c_re...  
61    www.104.com.tw/job/7skor?jobsource=jolist_c_re...  
...                                                 ...  
2760  www.104.com.tw/job/7qprq?jobsource=jolist_a_re...  
2784  www.104.com.tw/job/77lkz?jobsource=jolist_a_re...  
2801  www.104.com.tw/job/7sq3r?jobsource=jolist_b_re...  
2818  www.104.com.tw/job/7sq3r?jobsource=jolist_c_re...  
2844  www.104.com.tw/job/7rzj8?jobsource=jolist_d_re...  

[115 rows x 9 columns]

* #### 儲存檔案到google drive ####

In [ ]:
#只取得xlsx檔案路徑
glob.glob('jobs_csv/*.xlsx')

['jobs_csv/2022-11-27104職缺-數據分析.xlsx']